crates/astria-sequencer/src/app/storage/values/block_timestamp.rs

```diff
@@ -0,0 +1,94 @@
+use std::fmt::{
+    self,
+    Display,
+    Formatter,
+};
+
+use astria_eyre::eyre::bail;
+use borsh::{
+    io::{
+        Read,
+        Write,
+    },
+    BorshDeserialize,
+    BorshSerialize,
+};
+
+use super::{
+    Value,
+    ValueImpl,
+};
+
+#[derive(Debug)]
+pub(in crate::app) struct BlockTimestamp(tendermint::time::Time);
```

Comment:
General note: I wish we didn't use the tendermint time but instead our own type.